clients/include/testing_getrs.hpp

```diff
@@ -35,6 +36,135 @@ inline void testname_getrs(const Arguments& arg, std::string& name)
+{
+    auto hipblasGetrsFn = arg.fortran ? hipblasGetrs<T, true> : hipblasGetrs<T, false>;
+
+    hipblasLocalHandle handle(arg);
+    const int          N         = 100;
+    const int          nrhs      = 1;
+    const int          lda       = 101;
+    const int          ldb       = 102;
+    const size_t       A_size    = size_t(N) * lda;
+    const size_t       B_size    = ldb;
+    const int          Ipiv_size = N;
+
+    const hipblasOperation_t op = HIPBLAS_OP_N;
+
+    host_vector<T>   hA(A_size);
+    host_vector<T>   hB(B_size);
+    host_vector<T>   hX(B_size);
+    host_vector<int> hIpiv(Ipiv_size);
+
+    device_vector<T>   dA(A_size);
+    device_vector<T>   dB(B_size);
+    device_vector<int> dIpiv(Ipiv_size);
+    int                info = 0;
+
+    // Need initialization code because even with bad params we call roc/cu-solver
+    // so want to give reasonable data
+    EXPECT_HIPBLAS_STATUS(setup_getrs_testing(hA, hB, hX, hIpiv, dA, dB, dIpiv, N, lda, ldb),
+                          HIPBLAS_STATUS_SUCCESS);
+
+    EXPECT_HIPBLAS_STATUS(hipblasGetrsFn(handle, op, N, nrhs, dA, lda, dIpiv, dB, ldb, nullptr),
+                          HIPBLAS_STATUS_INVALID_VALUE);
+
+    EXPECT_HIPBLAS_STATUS(hipblasGetrsFn(handle, op, -1, nrhs, dA, lda, dIpiv, dB, ldb, &info),
+                          HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-2, info);
+
+    EXPECT_HIPBLAS_STATUS(hipblasGetrsFn(handle, op, N, -1, dA, lda, dIpiv, dB, ldb, &info),
+                          HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-3, info);
+
+    EXPECT_HIPBLAS_STATUS(hipblasGetrsFn(handle, op, N, nrhs, nullptr, lda, dIpiv, dB, ldb, &info),
+                          HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-4, info);
+
+    EXPECT_HIPBLAS_STATUS(hipblasGetrsFn(handle, op, N, nrhs, dA, N - 1, dIpiv, dB, ldb, &info),
+                          HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-5, info);
+
+    EXPECT_HIPBLAS_STATUS(hipblasGetrsFn(handle, op, N, nrhs, dA, lda, nullptr, dB, ldb, &info),
+                          HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-6, info);
+
+    EXPECT_HIPBLAS_STATUS(hipblasGetrsFn(handle, op, N, nrhs, dA, lda, dIpiv, nullptr, ldb, &info),
+                          HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-7, info);
+
+    EXPECT_HIPBLAS_STATUS(hipblasGetrsFn(handle, op, N, nrhs, dA, lda, dIpiv, dB, N - 1, &info),
+                          HIPBLAS_STATUS_INVALID_VALUE);
+    EXPECT_EQ(-8, info);
+
+    // If N == 0, A, B, and ipiv can be nullptr
+    EXPECT_HIPBLAS_STATUS(
+        hipblasGetrsFn(handle, op, 0, nrhs, nullptr, lda, nullptr, nullptr, ldb, &info),
+        HIPBLAS_STATUS_SUCCESS);
+    EXPECT_EQ(0, info);
+
+    // if nrhs == 0, B can be nullptr
+    EXPECT_HIPBLAS_STATUS(hipblasGetrsFn(handle, op, 0, nrhs, dA, lda, dIpiv, nullptr, ldb, &info),
+                          HIPBLAS_STATUS_SUCCESS);
+    EXPECT_EQ(0, info);
```

Comment:
This has N = 0 instead of nrhs = 0. (Same for other getrs tests)

Reply:
Fixed, will wait to see if it passes on cublas.